### PR TITLE
Prepare new release.

### PR DESCRIPTION
**Pull request by joerg84**

### CODE REVIEW

File: marathon/dev-jupyter-cni-notebook-marathon.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "container": {
     "type": "MESOS",
     "docker": {
-      "image": "dcoslabs/dcos-jupyter:latest",
+      "image": "dcoslabs/dcos-jupyterlab:latest",
```
Comment (joerg84) [inline marathon/dev-jupyter-cni-notebook-marathon.json]: Just a question of consistency: some of the docker images (e.g., in the dockerfiles above) are in the mesosphere org, while other in the dcoslabs. IMO if this is a beta product it should be in the mesosphere org.

File: marathon/dev-jupyter-cni-notebook-marathon.json [hunk 2]
```diff
@@ -146,7 +145,7 @@
     "SPARK_CONF_EVENTLOG_ENABLED": "spark.eventLog.enabled=true",
     "SPARK_CONF_EVENTLOG_DIR": "spark.eventLog.dir=hdfs://hdfs/history",
     "SPARK_CONF_HADOOP_FS_S3A_AWS_CREDENTIALS_PROVIDER": "spark.hadoop.fs.s3a.aws.credentials.provider=com.amazonaws.auth.InstanceProfileCredentialsProvider",
-    "SPARK_CONF_MESOS_EXECUTOR_DOCKER_IMAGE": "spark.mesos.executor.docker.image=dcoslabs/dcos-spark:latest",
+    "SPARK_CONF_MESOS_EXECUTOR_DOCKER_IMAGE": "spark.mesos.executor.docker.image=dcoslabs/dcos-jupyterlab:latest",
```
Comment (joerg84) [inline marathon/dev-jupyter-cni-notebook-marathon.json]: same here (mesosphere gh org)

File: marathon/dev-jupyter-notebook-marathon.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   "container": {
     "type": "MESOS",
     "docker": {
-      "image": "dcoslabs/dcos-jupyter:latest",
+      "image": "dcoslabs/dcos-jupyterlab:latest",
```
Comment (joerg84) [inline marathon/dev-jupyter-notebook-marathon.json]: dito (mesosphere GH org)

File: marathon/dev-jupyter-notebook-marathon.json [hunk 4]
```diff
@@ -66,7 +65,7 @@
     "SPARK_CONF_EVENTLOG_ENABLED": "spark.eventLog.enabled=true",
     "SPARK_CONF_EVENTLOG_DIR": "spark.eventLog.dir=hdfs://hdfs/history",
     "SPARK_CONF_HADOOP_FS_S3A_AWS_CREDENTIALS_PROVIDER": "spark.hadoop.fs.s3a.aws.credentials.provider=com.amazonaws.auth.InstanceProfileCredentialsProvider",
-    "SPARK_CONF_MESOS_EXECUTOR_DOCKER_IMAGE": "spark.mesos.executor.docker.image=dcoslabs/dcos-spark:latest",
+    "SPARK_CONF_MESOS_EXECUTOR_DOCKER_IMAGE": "spark.mesos.executor.docker.image=dcoslabs/dcos-jupyterlab:latest",
```
Comment (joerg84) [inline marathon/dev-jupyter-notebook-marathon.json]: dito (mesosphere gh org)

File: notebooks/TFoS.ipynb [hunk 5]
```diff
@@ -32,7 +32,7 @@
    "source": [
     "# CPU Config\n",
     "conf = SparkConf().setAppName('Mnist') \\\n",
-    "                          .set('spark.mesos.executor.docker.image', 'dcoslabs/dcos-spark:1.11.4-2.2.1') \n"
+    "                  .set('spark.mesos.executor.docker.image', 'dcoslabs/dcos-jupyterlab:latest') \n"
```
Comment (joerg84) [inline notebooks/TFoS.ipynb]: dito (mesosphere gh org)

File: notebooks/TFoS.ipynb [hunk 6]
```diff
@@ -45,9 +45,9 @@
    "source": [
     "# GPU Config\n",
     "#conf = SparkConf().setAppName('Mnist-GPU') \\\n",
-    "#                          .set('spark.mesos.executor.docker.image', 'dcoslabs/dcos-spark:1.11.4-2.2.1-gpu') \n",
-    "#                          .set('spark.mesos.gpus.max', 5) \\\n",
-    "#                          .set('spark.mesos.executor.gpus', 1)"
+    "#                  .set('spark.mesos.executor.docker.image', 'dcoslabs/dcos-jupyterlab:latest-gpu') \\\n",
```
Comment (joerg84) [inline notebooks/TFoS.ipynb]: dito (mesosphere gh org)

File: CHANGELOG.md [hunk 7]
```diff
@@ -47,7 +77,11 @@
...
-* Rename `OIDC_AUTH_METHOD` to `OIDC_TOKEN_ENDPOINT_AUTH_METHOD` to disambiguate from the Introspection Endpoint Authentication method
+* The `OIDC_REDIRECT_URI` environment variable must now be specified as an absolute URI since [redirect_uri_path is deprecated](https://github.com/zmartzone/lua-resty-openidc/commit/0f2a68b82cf4849fc3efe4b25c389fc45377fc63)
+* Rename the `OIDC_AUTH_METHOD` environment variable to `OIDC_TOKEN_ENDPOINT_AUTH_METHOD` to disambiguate from the Introspection Endpoint Authentication method
+
+### Features
 
+* Apache Toree, as of [0.3.0-incubating-rc1](https://github.com/apache/incubator-toree/releases/tag/v0.3.0-incubating-rc1) has [removed support for PySpark and SparkR](https://github.com/apache/incubator-toree/commit/276165ae2ac136a59d208058a031caf769bb312e), only the Scala and SQL interpreters will remain available. The [vanilla PySpark and SparkR kernels, however retain their ability to launch pre-configured Spark Jobs](https://github.com/dcos-labs/dcos-jupyterlab-service/blob/master/jupyter_notebook_config.py#L231-L232)
```
[the remark this answers is not in the document]
Comment (vishnu2kmohan) [reply CHANGELOG.md]: It's a breaking change in an upstream project. 

Under which sub-section of `Breaking Changes` would you like to have it sit?

File: Dockerfile [hunk 8]
```diff
@@ -24,24 +23,25 @@ ARG HADOOP_MAJOR_VERSION="2.9"
 ARG HADOOP_SHA256="eed6015a123644d3b4247bac58770e4a8b31340fa62721987430e15a0dd942fc"
 ARG HADOOP_URL="http://www-us.apache.org/dist/hadoop/common"
 ARG HADOOP_VERSION="2.9.1"
-ARG HOME="/home/beakerx"
+ARG HOME="/home/jovyan"
```
[the remark this answers is not in the document]
Comment (vishnu2kmohan) [reply Dockerfile]: It's what the upstream Jupyter peeps use.

File: Dockerfile [hunk 9]
```diff
@@ -98,7 +98,7 @@ ENV BOOTSTRAP="${MESOSPHERE_PREFIX}/bin/bootstrap" \
     MESOS_NATIVE_JAVA_LIBRARY="${MESOSPHERE_PREFIX}/libmesos-bundle/lib/libmesos.so" \
     NB_GID=${NB_GID:-"100"} \
     NB_UID=${NB_UID:-"1000"} \
-    NB_USER=${NB_USER:-"beakerx"} \
+    NB_USER=${NB_USER:-"jovyan"} \
```
[the remark this answers is not in the document]
Comment (vishnu2kmohan) [reply Dockerfile]: `NB_USER=${NB_USER:-"jovyan"}` is valid bash (which works within the `Dockerfile`)?

File: Dockerfile-data-toolkit [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+ARG TENSORFLOW_URL="https://storage.googleapis.com/tensorflow"
+ARG TENSORFLOW_VARIANT="cpu"
+ARG TENSORFLOW_VERSION="1.11.0"
+ARG TINI_GPG_KEY="595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7"
+ARG TINI_URL="https://github.com/krallin/tini/releases/download"
+ARG TINI_VERSION="v0.18.0"
+ARG VCS_REF
+ARG XGBOOST_JAVA_JAR_SHA256="4a6599ee3f1bd10d984e8b03747d5bc3cb637aeb791474178de2c285857bf69e"
+ARG XGBOOST_SPARK_JAR_SHA256="cd31fb96b26fee197e126215949bc4f5c9a3cafd7ff157ab0037a63777c2935e"
+ARG XGBOOST_URL="https://s3.amazonaws.com/vishnu-mohan/xgboost"
```
[the remark this answers is not in the document]
Comment (vishnu2kmohan) [reply Dockerfile-data-toolkit]: Yep, I don't have privileges. We need CI.

File: marathon/dev-jupyter-cni-notebook-marathon.json (hunk 2 above)
Comment (vishnu2kmohan) [reply marathon/dev-jupyter-cni-notebook-marathon.json]: Privileges/Credentials and CI needed.

File: jupyter_notebook_config.py [hunk 11]
```diff
@@ -127,7 +130,7 @@
     else:
         spark_opts.append('--files={}'.format(os.getenv('KRB5CCNAME')))
     spark_opts.append('--conf spark.executorEnv.KRB5CCNAME={}/krb5cc_{}'.format(
-        os.getenv('MESOS_SANDBOX'), os.getuid()))
+        os.getenv('MESOS_SANDBOX', '/home/jovyan'), os.getuid()))
```
[the remark this answers is not in the document]
Comment (vishnu2kmohan) [reply jupyter_notebook_config.py]: Trying to remember why I used `/home/jovyan` explicitly, but you're right, I could use `${HOME}` to be more generic over just `"/home/${NB_USER}"`

File: marathon/dev-jupyter-notebook-marathon.json (hunk 3 above)
Comment (vishnu2kmohan) [reply marathon/dev-jupyter-notebook-marathon.json]: CI

File: marathon/dev-jupyter-notebook-marathon.json (hunk 4 above)
Comment (vishnu2kmohan) [reply marathon/dev-jupyter-notebook-marathon.json]: CI

File: notebooks/TFoS.ipynb (hunk 5 above)
Comment (vishnu2kmohan) [reply notebooks/TFoS.ipynb]: CI

File: notebooks/TFoS.ipynb (hunk 6 above)
Comment (vishnu2kmohan) [reply notebooks/TFoS.ipynb]: CI

File: Dockerfile (hunk 8 above)
[the remark this answers is not in the document]
Comment (joerg84) [reply Dockerfile]: +1

File: Dockerfile (hunk 9 above)
[the remark this answers is not in the document]
Comment (joerg84) [reply Dockerfile]: I wanted to reduce the hardcoded places where we use jovyan.

File: Dockerfile-data-toolkit (hunk 10 above)
[the remark this answers is not in the document]
Comment (joerg84) [reply Dockerfile-data-toolkit]: I can move.

**Comment by joerg84** [comment]
dcoslabs/dcos-jupyterlab:latest -> mesosphere/dcos-jupyterlab:1.3.0-0.35.2

**Comment by joerg84** [comment]
LGTM @vishnu2kmohan

**Pull request merged**